### PR TITLE
Feature/152 email change

### LINKED ISSUE #152: Email change with verify

### PR DESCRIPTION
**Pull request by coffemakingtoaster**
# Changes
- add mail change functionality

Closes #152

### CODE REVIEW

**Comment by coffemakingtoaster** [comment]
TODO: Verify that email is not already in db

File: backend/src/db/services/user.service.ts [hunk 1]
```diff
@@ -90,4 +90,18 @@ export class UserService {
...
+
+  async updateUserEmail(id: ObjectId, email: string) {
+    try {
+      await this.userModel.updateOne(
+        { _id: id },
+        { email, hasVerifiedEmail: false },
+      )
+    } catch (error) {
+      this.logger.error(error)
+      // Necessary due to incomplete typeorm type
```
Comment (Branchverse) [inline backend/src/db/services/user.service.ts:102]: this is not typeorm

File: backend/test/profile.e2e-spec.ts [hunk 2]
```diff
@@ -152,4 +158,174 @@ describe('AuthController (e2e)', () => {
...
+        const sendMail = mock.getSentMail()[0]
+        const verifyToken = getVerifyTokenFromMail(sendMail.html as string)
+        const tokenPayload: VerifyJWTPayload = jwtService.verify(verifyToken, {
+          secret: configService.get<string>('JWT_VERIFY_SECRET'),
+        })
+
+        expect(tokenPayload.email).not.toEqual(SAMPLE_USER.email)
+        expect(tokenPayload.email).toEqual('newmail@mail.mail')
+      })
+    })
```
Comment (Branchverse) [inline backend/test/profile.e2e-spec.ts:278]:
```suggestion
    describe('Positive Tests', () => {
    const newMail = 'newmail@mail.mail'
      it('should set new email', async () => {
        // ACT
        const res = await request(app.getHttpServer())
          .patch('/profile/change-email')
          .send({
            email: newMail ,
          })
          .set({
            Authorization: `Bearer ${token}`,
          })
        // ASSERT
        expect(res.statusCode).toEqual(HttpStatus.OK)
        const oldUser = await userModel.findOne({
          email: SAMPLE_USER.email,
        })
        expect(oldUser).toBeNull()
        
        const updatedUser = await userModel.findOne({
          email: newMail ,
        })
        expect(updatedUser).toBeDefined()
      })

      it('should set user mail as not verified', async () => {
        // ARRANGE
        await userModel.updateOne(
          { email: SAMPLE_USER.email },
          { hasVerifiedEmail: true },
        )
        // ACT
        const res = await request(app.getHttpServer())
          .patch('/profile/change-email')
          .send({
             email: newMail ,
          })
          .set({
            Authorization: `Bearer ${token}`,
          })
        // ASSERT
        expect(res.statusCode).toEqual(HttpStatus.OK)
        const updatedUser = await userModel.findOne({
           email: newMail ,
        })
        expect(updatedUser.hasVerifiedEmail).toEqual(false)
      })

      it('should send verify email', async () => {
        // ACT
        const res = await request(app.getHttpServer())
          .patch('/profile/change-email')
          .send({
            email: newMail ,
          })
          .set({
            Authorization: `Bearer ${token}`,
          })
        // ASSERT
        expect(res.statusCode).toEqual(HttpStatus.OK)
        expect(mock.getSentMail().length).toEqual(1)
      })

      it('should not send verify email if provided email is not new', async () => {
        // ACT
        const res = await request(app.getHttpServer())
          .patch('/profile/change-email')
          .send({
            email: SAMPLE_USER.email,
          })
          .set({
            Authorization: `Bearer ${token}`,
          })
        // ASSERT
        expect(res.statusCode).toEqual(HttpStatus.OK)
        expect(mock.getSentMail().length).toEqual(0)
      })

      it('should include correct token in verify mail', async () => {
        // ACT
        const res = await request(app.getHttpServer())
          .patch('/profile/change-email')
          .send({
            email: newMail ,
          })
          .set({
            Authorization: `Bearer ${token}`,
          })
        // ASSERT
        expect(res.statusCode).toEqual(HttpStatus.OK)

        const sendMail = mock.getSentMail()[0]
        const verifyToken = getVerifyTokenFromMail(sendMail.html as string)
        const tokenPayload: VerifyJWTPayload = jwtService.verify(verifyToken, {
          secret: configService.get<string>('JWT_VERIFY_SECRET'),
        })

        expect(tokenPayload.email).not.toEqual(SAMPLE_USER.email)
        expect(tokenPayload.email).toEqual(newMail)
      })
    })
```

**Review by Branchverse — APPROVED** [review]
lgtm,
pls fix minor changes

**Pull request merged**